timemachine/fe/free_energy.py

```diff
@@ -496,16 +496,9 @@ def _combine(self, ligand_values, host_values=None):
 def get_water_sampler_params(initial_state: InitialState) -> NDArray:
     """Given an initial state, return the parameters that define the nonbonded parameters of water with respect to the
     entire system.
-
-    Since we split the different components of the NB into ligand-water, ligand-protein, we want to use the ligand-water parameter
-    to ensure that the ligand component is correctly configured for the specific lambda window.
     """
-    summed_pot = get_bound_potential_by_type(initial_state.potentials, SummedPotential).potential
-    # TBD Figure out a better way of handling the jankyness of having to select the IxnGroup that defines the Ligand-Water
```

Comment:
@badisa  Can you confirm passing in the full NB ixn params (L-W and L-P) to the water sampler is the correct thing to do here? Since we don't split the ixns by type anymore.

Reply:
Hmmmm, would probably be wrong actually. Since the goal of the L-W/P potential is to adjust the water and protein parameters with respect to the Ligand. And currently where we adjust L-W, but not L-P, we are fine since we are moving waters with respect to the ligand. Once we mix them together, moving the waters will be influenced by the changes to L-P since the water will see the protein parameters.

But that is true from https://github.com/proteneer/timemachine/pull/1372 and this PR seems like strictly an improvement. We probably need to move away from this janky behavior where we assume that there is a single potential that contains the right parameters and instead construct a set of water and protein params that are the global state and adjust the ligand parameters independently.

Reply:
Updated the code to handle this and added a test in test_free_energy.py::test_get_water_sampler_params_complex